### DISCUSSION TITLE
Ability for dialog root to be rendered anywhere in DOM tree

### DISCUSSION THREAD

**Discussion by ustyantsevxx** [tailwindlabs/headlessui#833] [Ideas]
Right now dialog always teleports to the end of `body`. In some cases it might be critical to render it inside of some other element. E.g delete confirmation prompt in a drawer (dialog inside of dialog).

Basically, it would be cool if dialog had some prop that disables behavior described below 😀 (docs fragment):
> So, when using our Dialog, there's no need to use a Portal yourself! We've already taken care of it.

**Comment by RobinMalfait** [comment]
If you nest your dialogs, then the root is already taken care of properly.
